tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/connection/v3/OpenNewConnectionWizardTest.java

```diff
@@ -28,50 +27,44 @@
  * Test opening new connection dialog via tool item and context menu.
  * 
  * @author mlabuda@redhat.com
+ * @author adietish@redhat.com
  *
  */
-public class OpenNewConnectionShellTest {
+public class OpenNewConnectionWizardTest {
 	
 	OpenShiftExplorerView explorer = new OpenShiftExplorerView();
 	
 	@Test
-	public void testOpenConnectionShellViaLinkTest() {
+	public void shouldOpenConnectionWizardViaLinkTest() {
+		// given
+		ConnectionUtils.clearAll();
 		explorer.open();
-		
-		try {
-			new DefaultLink(OpenShiftLabel.TextLabels.NEW_CONNECTION).click();
-			// PASS
-		} catch (SWTLayerException ex) {
-			fail("New Connection shell cannot be opened via context menu.");
-		}
-		
-		verifyNewConnectionShellIsOpened();
+
+		// when
+		new DefaultLink(OpenShiftLabel.TextLabels.CREATE_CONNECTION).click();
+
+		// then
+		verifyNewConnectionWizardIsOpened();
 	}
 	
 	@Test
-	public void testOpenConnectionShellViaToolItemTest() {
+	public void shouldOpenConnectionWizardViaToolItemTest() {
+		// given
 		explorer.open();
 		
-		try {
-			new DefaultToolItem(OpenShiftLabel.Others.CONNECT_TOOL_ITEM).click();
-			// PASS
-		} catch (SWTLayerException ex) {
-			fail("New Connection shell cannot be opened via view tool item.");
-		}
-		
-		verifyNewConnectionShellIsOpened();
+		// when
+		new DefaultToolItem(OpenShiftLabel.Others.CONNECT_TOOL_ITEM).click();
+
+		// then
+		verifyNewConnectionWizardIsOpened();
 	}
 	
-	private void verifyNewConnectionShellIsOpened() {
-		Shell connectionShell = null;
-		try {
-			connectionShell = new DefaultShell(OpenShiftLabel.Shell.NEW_CONNECTION);
-		} catch (SWTLayerException ex) {
-			fail("New Connection shell was not opened.");
-		}
-		
+	private void verifyNewConnectionWizardIsOpened() {
```

Comment:
@mlabuda @rhopp I still dont understand why for you it makes a difference if a test is failing or has an exception. Maven does this distinction, and so does Eclipse JUnit view too. But then both cases are treated as failing tests, there's no real difference in this sense. What do I miss?
Then generally speaking ```try/catch(reddeer exception) { fail() }``` is an anti-pattern as I tried to point out. It hides the real cause and replace it with a junit fail. Adding the stacktrace in text helps, but is imho not equivalent. See here for further details:
* http://www.kyleblaney.com/junit-best-practices/ ("Do not write your own catch blocks that exist only to fail a test")
* http://stackoverflow.com/questions/31423643/try-catch-in-a-junit-test ("")
Generally speaking I would also emphasize that if you are repeating a pattern - like your  try/catch/fail - in user code, then there are great indications that this should be done either in util or framework. I thus think that if you're still not happy with the red deer exception being thrown when a widget is not found, then you should change reddeer.